File: content-solutions/stack/ribbon-tags.js
function solution(tags) {
  const stack = [];
  for (const tag of tags) {
    const closing = tag[1] === "/";
    const name = closing ? tag.slice(2, -1) : tag.slice(1, -1);
    if (closing) {
      if (stack.length === 0 || stack.pop() !== name) return false;
    } else {
      stack.push(name);
    }
  }
  return stack.length === 0;
}

File: content-solutions/stack/ribbon-tags.py
def solution(tags):
    stack = []
    for tag in tags:
        closing = tag[1] == "/"
        name = tag[2:-1] if closing else tag[1:-1]
        if closing:
            if not stack or stack.pop() != name:
                return False
        else:
            stack.append(name)
    return not stack
